tools/SourceKit/tools/sourcekitd/lib/API/CMakeLists.txt

```diff
@@ -14,10 +18,14 @@ set(sourcekitdAPI_NonDarwin_InProc_sources
   sourcekitdAPI-InProc.cpp)
 set(LLVM_OPTIONAL_SOURCES ${sourcekitdAPI_Darwin_sources} ${sourcekitdAPI_NonDarwin_InProc_sources})
 
-if(APPLE)
+if(APPLE AND HAVE_XPC_H)
   list(APPEND sourcekitdAPI_sources ${sourcekitdAPI_Darwin_sources})
 elseif(SWIFT_SOURCEKIT_USE_INPROC_LIBRARY)
   list(APPEND sourcekitdAPI_sources ${sourcekitdAPI_NonDarwin_InProc_sources})
+elseif(APPLE)
+  # when XPC isn't available
+  set(BUILD_SOURCEKIT_USE_INPROC_LIBRARY TRUE)
```

Comment:
Maybe SWIFT_SOURCEKIT_USE_INPROC_LIBRARY should just come first, and then if that's not set we can check APPLE and error if !HAVE_XPC_H.